modules/javafx.graphics/src/main/java/com/sun/pisces/JavaSurface.java

```diff
@@ -43,6 +43,7 @@ public JavaSurface(int[] dataInt, int dataType, int width, int height) {
         this.dataBuffer = IntBuffer.wrap(this.dataInt);
 
         initialize(dataType, width, height);
+        addDisposerRecord();
```

Comment:
Should this be called from the superclass instead? It works as-is, but if there were ever another subclass added, it would have to be replicated there as well.

Reply:
`AbstractSurface.nativePtr` is needed to create the `AbstractSurfaceDisposerRecord`.
It is created and set by the native method `initialize(dataType, width, height);` invoked at line 45 in the `JavaSurface` constructor.
It is the reason `addDisposerRecord()` is needed here.

Reply:
Oh, I see. The native initialize method in the subclass is writing into a private field in the superclass. Can you add a comment to this effect, since it isn't obvious without reading the native code?

Reply:
The comment would be really helpful for readers.